### PR TITLE
Vue unit tests

### PR DESCRIPTION
**Pull request by bafolts**
This change will use actual vue unit test tools instead of the rudimentary ones I had in place.

https://vue-test-utils.vuejs.org/

This will allow us to examine what happens with the components during events and state changes.

### CODE REVIEW

File: tests/components/TagCount.spec.ts [hunk 1]
```diff
@@ -1,35 +1,32 @@
...
+        return localVue;
+    }
+    it("renders with no count", function () {
+        const tagCount = mount(TagCount, {
+            localVue: getLocalVue(),
+            propsData: {
+                count: 0
+            }
+        });
+        expect(tagCount.find("div[class=\"tag-display tag-no-show\"]").exists()).to.eq(true);
```
Comment (kberg) [inline tests/components/TagCount.spec.ts:19]: s/to.eq(true)/is.true

Comment (kberg) [inline tests/components/TagCount.spec.ts:19]: Good for now, let's consider https://www.chaijs.com/plugins/chai-dom/ later? Perhaps that's one I can dop.

**Review by kberg — APPROVED** [review]

**Comment by kberg** [comment]
This is great!

File: tests/components/TagCount.spec.ts (hunk 1 above)
Comment (bafolts) [reply tests/components/TagCount.spec.ts:19]: The heavy lifting with getting jsdom in place and the vue-test-utils running is in place with this change. Should be easy for others to drop in updates later. I am not sure `chai-dom` would work. The `Wrapper` exposes these `find` methods.

https://vue-test-utils.vuejs.org/api/wrapper/#find

As `find` is deprecated I will need to update these in some way to leverage `findComponent`.

**Pull request merged**